### PR TITLE
[REVIEW] Slice a column into multiple vectors

### PR DESCRIPTION
**Pull request by razajafri**
This API slices a column vector into multiple column vectors based on a set of indices passed. The indices can be an int array or a column vector of ints.

### CODE REVIEW

**Comment by revans2** [comment]
Can we please see some tests for what happens if I pass in an odd number of entries into the slice column?  Also what happens if my indexes are out of bounds?  Do we just walk off into other places in memory?

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 1]
```diff
@@ -301,4 +303,82 @@ void testQuantilesOnDoubleInput() {
       }
     }
   }
+
+  @Test
+  void testSliceWithArray() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    try(ColumnVector cv = ColumnVector.fromBoxedInts(10, 12, null, null, 18, 20, 22, 24, 26, 28)) {
+      //{{12, null}, {20, 22, 24, 26}, {14, 16}, {}}
```
Comment (nartal1) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: Nit: This line can be removed. Same comment for other test as well.

**Comment by razajafri** [comment]
> Can we please see some tests for what happens if I pass in an odd number of entries into the slice column? Also what happens if my indexes are out of bounds? Do we just walk off into other places in memory?

In case the indices are odd, it throws a cudfException. I have added the test for that. 
In case any of the indices are out of range the result is undefined as noted in the function javadocs

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java (hunk 1 above)
Comment (razajafri) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: PTAL

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public Table(ColumnVector... columns) {
     nativeHandle = createCudfTable(cudfColumnPointers);
   }
 
-  private Table(long[] cudfColumns) {
+  Table(long[] cudfColumns) {
```
[the remark this answers is not in the document]
Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/Table.java]: Updated PTAL

**Pull request merged**